File: CWM.UI/cwm_desktop_mobile/lib/models/part_work_order.dart
import 'package:cwm_desktop_mobile/models/part.dart';
import 'package:cwm_desktop_mobile/models/vehicle.dart';
import 'package:cwm_desktop_mobile/models/work_order.dart';
import 'package:json_annotation/json_annotation.dart';

part 'part_work_order.g.dart';

@JsonSerializable()
class PartWorkOrder {
  int id;
  DateTime serviceDate;
  Vehicle? vehicle;
  WorkOrder? workOrder;
  Part? part;

  PartWorkOrder(
    this.id,
    this.serviceDate,
    this.vehicle,
    this.workOrder,
    this.part,
  );

  factory PartWorkOrder.fromJson(Map<String, dynamic> json) =>
      _$PartWorkOrderFromJson(json);

  Map<String, dynamic> toJson() => _$PartWorkOrderToJson(this);
}

File: CWM.Model/Models/PartWorkOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CWM.Core.Models
{
    public class PartWorkOrder : Base
    {
        public DateTime? ServiceDate { get; set; }
        public Vehicle? Vehicle { get; set; } = new();
        public Part? Part { get; set; } = new();
        public WorkOrder? WorkOrder { get; set; } = new();
    }
}
